app/assets/javascripts/host_power.js

```diff
@@ -0,0 +1,61 @@
+(function(){
+
+  $(document).ready(function () {
+    powerStatus();
+  });
+
+  function hostIds() {
+    var res = []
+    $('#hosts-table input.host_select_boxes').each(function (index, item) {
+      res.push($(item).attr('id').match(/\d+/).pop());
+    });
+    return res;
+  }
+
+  function findLabel(id) {
+     var label = null;
+     $('#hosts-table input.host_select_boxes').each(function (index, item) {
+        if ($(item).attr('id').match(/\d+/).pop() === id) {
+          label = $(item).parent().next().children(":first");
```

Comment:
Something like `return $('#hosts-table input.host_select_boxes #host_ids_'+id).parent().next().children(':first');` should get the same result if I understand correctly.
Another way to reduce the js complexity in general would be to give each row an id that is the same as the `host.id`, instead of relying on the unrelated checkbox, parsing its id, and traversing the DOM from it, as the DOM tree may change in the future.